Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/RequestsManager.java

```diff
@@ -19,16 +29,85 @@
     /* (non-Javadoc)
      * @see edu.asu.conceptpower.app.core.IRequestsManager#addReviewRequest(edu.asu.conceptpower.core.ReviewRequest)
      */
-    public void addReviewRequest(ReviewRequest newReviewRequest) {
+    public void addReviewRequest(ReviewRequest reviewRequest) {
+        reviewRequest.setId(generateId(REVIEW_PREFIX));
+        dbClient.store(reviewRequest);
+    }
+    
+    /* (non-Javadoc)
+     * @see edu.asu.conceptpower.app.core.IRequestsManager#getLatestReview(java.lang.String)
+     */
+    public ReviewRequest getLatestReview(String conceptId) {
+        List<ReviewRequest> reviewRequests = getAllReviews(conceptId);
+       
+        if(reviewRequests!= null && reviewRequests.size() > 0) {
```

Comment:
Is "reviewRequests.size() > 0" check required here? getAllReviews either returns list with more than 0 elements  or it will return null.

Reply:
We may not be sure. Just to keep the code defensive, it's good to check for null and empty cases. Instead of `reviewRequests.size() > 0` we can actually use inbuilt method (`reviewRequests.isEmpty()`) to check for empty case.